python/cuml/fil/fil.pyx

```diff
@@ -104,6 +104,12 @@ cdef class TreeliteModel():
         TreeliteQueryNumFeature(self.handle, &out)
         return out
 
+    @staticmethod
+    def free_treelite_model(model_handle):
+        cdef uintptr_t model_ptr = <uintptr_t>model_handle
+        TreeliteFreeModel(<ModelHandle> model_ptr)
+        del model_handle
```

Comment:
I believe this del is unnecessary and maybe a little confusing... the "handle" still exists, it's just the model pointed ot by the handle that we're wiping out here.